cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -61,6 +71,52 @@ def transfer_flow_cell(
 
         LOG.info(f"Flow cell added: {flow_cell}")
 
+    def add_sample_lane_sequencing_metrics_for_flow_cell(self):
+        """Add sample lane sequencing metrics to status database."""
+        flow_cell_dir: Path = self.flow_cell_dir
+
+        metrics = create_sample_lane_sequencing_metrics_for_flow_cell(
+            flow_cell_dir=flow_cell_dir,
+            bcl_converter=self.infer_bcl_converter(),
+        )
+
+        self.status_db.session.add_all(metrics)
+        self.status_db.session.commit()
+        LOG.info(
+            f"Added sample lane sequencing metrics to status database for: {self.flow_cell_name}"
+        )
+
+    def finish_flow_cell_temp(self):
+        """Finish flow cell."""
+        self.add_sample_lane_sequencing_metrics_for_flow_cell()
+
+    def infer_bcl_converter(self) -> str:
+        """Set bcl converter from flow cell."""
+        if self.is_bcl2fastq_demux_folder_structure():
+            LOG.info("Flow cell is demultiplexed with bcl2fastq")
+            return "bcl2fastq"
+        LOG.info("Flow cell is demultiplexed with bcl_converter")
+        return "bcl_convert"
+
+    def is_bcl2fastq_demux_folder_structure(self) -> bool:
+        """Check if flow cell directory is bcl2fastq demux folder structure."""
+        lane_tile_folder_pattern = r"l\dt\d{2}"
+        for folder in os.listdir(self.flow_cell_dir):
+            if re.search(lane_tile_folder_pattern, folder):
+                return True
```

Comment:
I don't know if this is what we want. This will return true if the first folder fulfils the condition and leave the rest of the folders unchecked.

Reply:
Indeed, the folder structure is dynamic. when one of these lt1t21 etc. folders is there it is a bcl2fastq output. sometimes this can be 4 lanes, sometimes 8. One present folder "should" be enough for the check to pass though. But we can revise on how we check the folder structure.

Reply:
Lets improve the check in a future PR